core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -323,15 +322,20 @@ public void serverStopped(Server server) throws Exception {
     @Override
     public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
         final long longPollingTimeoutMillis = getLongPollingTimeoutMillis(req);
-        final boolean isHealthy = isHealthy();
+        final HealthStatus healthStatus = healthStatus();
         final boolean useLongPolling;
         if (longPollingTimeoutMillis > 0) {
             final String expectedState =
                     Ascii.toLowerCase(req.headers().get(HttpHeaderNames.IF_NONE_MATCH, ""));
             if ("\"healthy\"".equals(expectedState) || "w/\"healthy\"".equals(expectedState)) {
-                useLongPolling = isHealthy;
+                useLongPolling = healthStatus == HealthStatus.HEALTHY;
```

Comment:
note: this change means if `HEALTHY` or `UNHEALTHY` isn't set, then long polling never works since armeria client only sends `healthy` or `unhealthy` depending on the status code.

https://github.com/line/armeria/blob/579a6d0998e34605c41ea13b26c1f7c19836e908/core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthChecker.java#L105-L106

Reply:
Oh, I see. Than I think we can change the logic as follows:

When the `expectedState` is
- `healthy`:  use long polling when the status is `HEALTHY` or `DEGRADED` 
- `unhealthy`:  use long polling when the status is `UNHEALTHY` or `UNDER_MAINTENANCE`

Reply:
I understood the objective of this issue to propagate the status to clients.
If we only maintain `expectedState`, I think clients may not receive the status on update for long polling health check requests.

If we would like to maintain backwards compatibility, we could define a separate syntax for status matching.

e.g.
- `If-None-Match=[healthy|unhealthy]`: propagates a signal whenever a status changes between healthy <-> unhealthy
- `If-None-Match=status=[HEALTHY|DEGRADED|UNHEALTHY|UNDER_MAINTENANCE|STOPPING]`: propagates a signal when the status does not match the status exactly

Reply:
I've added `If-None-Match=status=[HEALTHY|DEGRADED|UNHEALTHY|UNDER_MAINTENANCE|STOPPING]` and preserved `If-None-Match=[healthy|unhealthy]` in order to maintain backward compatibility. 

- When `If-None-Match=[healthy|unhealthy]` is present, return a response when the server becomes `healthy` or `unhealthy` 
- When `If-None-Match=status=[HEALTHY|DEGRADED|UNHEALTHY|UNDER_MAINTENANCE|STOPPING]` is present, return a response when the server's `HealthStatus` is changed.